src/compiler/parser/lines_8000_end.rs

```diff
@@ -0,0 +1,250 @@
+use std::cell::{Cell, Ref, RefCell};
+use std::convert::TryInto;
+use std::rc::Rc;
+
+use crate::{
+    file_extension_is_one_of, for_each_child_bool, Debug_, DiagnosticMessage, Extension,
+    IncrementalParserSyntaxCursorReparseTopLevelAwait, Node, NodeArray, NodeInterface, ParserType,
+    ReadonlyTextRange, TextChangeRange,
+};
+
+pub fn IncrementalParser() -> IncrementalParserType {
+    IncrementalParserType::new()
+}
+
+pub struct IncrementalParserType {}
+
+impl IncrementalParserType {
+    pub fn new() -> Self {
+        Self {}
+    }
+
+    pub fn update_source_file(
+        &self,
+        source_file: &Node, /*SourceFile*/
+        new_text: String,
+        text_change_range: TextChangeRange,
+        aggressive_checks: bool,
+    ) -> Rc<Node /*SourceFile*/> {
+        unimplemented!()
+    }
+
+    pub fn create_syntax_cursor(
+        &self,
+        source_file: &Node, /*SourceFile*/
+    ) -> IncrementalParserSyntaxCursor {
+        IncrementalParserSyntaxCursor::create(source_file.node_wrapper())
+    }
+}
+
+pub trait IncrementalParserSyntaxCursorInterface {
+    fn current_node(&self, parser: &ParserType, position: usize) -> Option<Rc<Node>>;
+}
+
+pub enum IncrementalParserSyntaxCursor {
```

Comment:
It looked like there aren't a lot of different implementations of `IncrementalParser.SyntaxCursor` so I modeled it as an enum